src/http/adapt.ml

```diff
@@ -74,7 +74,7 @@ let forward_body
     response
     (Httpun.Body.Writer.write_string body)
     (Httpun.Body.Writer.write_bigstring body)
-    (Httpun.Body.Writer.flush body)
+    (fun fn -> Httpun.Body.Writer.flush body (fun _ -> fn ()))
```

Comment:
`Httpun...flush` now passes `[ \`Closed | \`Written ]` to the function whereas previously it was just unit. It works if we ignore it but I don't think that's the best way to do it.